operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/ResourceController.java

```diff
@@ -7,27 +7,32 @@
 
   /**
    * The implementation should delete the associated component(s). Note that this is method is
-   * called when an object is marked for deletion. After its executed the custom resource finalizer
-   * is automatically removed by the framework; unless the return value is false - note that this is
-   * almost never the case. Its important to have the implementation also idempotent, in the current
-   * implementation to cover all edge cases actually will be executed mostly twice.
+   * called when an object is marked for deletion. After it's executed the custom resource finalizer
+   * is automatically removed by the framework; unless the return value is {@link
+   * DeleteControl#NO_FINALIZER_REMOVAL} - note that this is almost never the case. It's important
+   * to have the implementation also idempotent, in the current implementation to cover all edge
+   * cases actually will be executed mostly twice.
    *
    * @param resource
-   * @return true - so the finalizer is automatically removed after the call. false if you don't
-   *     want to remove the finalizer. Note that this is ALMOST NEVER the case.
+   * @return {@link DeleteControl#DEFAULT_DELETE} - so the finalizer is automatically removed after
+   *     the call. {@link DeleteControl#NO_FINALIZER_REMOVAL} if you don't want to remove the
+   *     finalizer. Note that this is ALMOST NEVER the case.
    */
-  DeleteControl deleteResource(R resource, Context<R> context);
+  default DeleteControl deleteResource(R resource, Context<R> context) {
```

Comment:
I approved the PR. LGTM, I think this part might be controversial, thus to not force the users to override this method.
I guess the reasoning here is that the deletion is usually handled using cascading delete.

Reply:
@adam-sandor what do you think?

Reply:
I'm on the fence here, so if @metacosm thinks this is good I'm ok with it. You're right that cascading delete will often be used.

Reply:
the main reason for this change was bases on the observation that most of the operator I saw so far rely on cascading delete so making the method `default` was about to match such observation.

Reply:
yeah, it makes sense from that perspectives, we managed lot's of external (not k8s) resources, there it does not. But this interface is small enough and well documented so I think it's fine this way. More like wanted to hear / document the arguments around this. thx!